log.cpp

```diff
@@ -0,0 +1,79 @@
+#include "log.hpp"
+#include <iostream>

```

Comment:
You should not import any standard libraries into Log.hpp or Log.cpp. All standard library imports go to Protocol.hpp.

1. Open Protocol.hpp, look at all the existing header files. If any of iostream, fstream, or ctime are not there, then add them.
2. #Include Protocol.hpp into Log.hpp

The only include in Log.cpp should be Log.hpp. The file will automatically inherit all definitions from the Protocol.hpp file if you've included it in Log.hpp.